azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,276 @@
+            const cliCredentials = await msRestAzure.AzureCliCredentials.create();
+            subscriptionID = cliCredentials.subscriptionInfo.id;
+            credentials = cliCredentials;
+        } else {
+            credentials = await msRestAzure.loginWithServicePrincipalSecret(clientID, clientSecret, tenantID);
+        }
+
+        return new cdnManagement.CdnManagementClient(credentials, subscriptionID);
+    }
+
+    async check(olds: DynamicProviderInputs, news: DynamicProviderInputs): Promise<pulumi.dynamic.CheckResult> {
+        const cdnClient = await this.getCDNManagementClient();
+        let result: pulumi.dynamic.CheckResult;
+
+        // If none of the CDN properties changed, then there is nothing to be validated.
+        if (olds.profileName === news.profileName &&
+            olds.endpointName === news.endpointName &&
+            olds.customDomainHostName === news.customDomainHostName) {
+            return {
+                inputs: news
+            };
+        }
+
+        const validationOutput = await cdnClient.endpoints.validateCustomDomain(
+            news.resourceGroupName, news.profileName, news.endpointName, news.customDomainHostName);
+        if (!validationOutput.customDomainValidated) {
+            result = {
+                failures: [
+                    {
+                        property: this.name,
+                        reason: validationOutput.reason || "domain_validation_failed"
+                    }
+                ]
+            };
+        } else {
+            result = {
+                inputs: news
+            };
+        }
+
+        return result;
+    }
+
+    async diff(id: string, previousOutput: DynamicProviderOutputs, news: DynamicProviderInputs): Promise<pulumi.dynamic.DiffResult> {
+        const replaces: string[] = [];
+        let changes = false;
+        let deleteBeforeReplace = false;
+
+        if (previousOutput.customDomainHostName !== news.customDomainHostName || previousOutput.name !== this.name) {
+            console.warn("Changing the domain name properties will cause a downtime.")
+
+            deleteBeforeReplace = true;
+            if (previousOutput.customDomainHostName !== news.customDomainHostName) {
+                replaces.push("customDomainHostName");
+            }
+            if (previousOutput.name !== this.name) {
+                replaces.push("customDomainName");
+            }
+        }
+
+        if (previousOutput.endpointName !== news.endpointName) {
+            deleteBeforeReplace = true;
+            replaces.push("endpointName");
+        }
+
+        // HTTPS can be enabled/disabled in-place.
+        if (previousOutput.httpsEnabled !== news.httpsEnabled) {
+            changes = true;
+            replaces.push("httpsEnabled");
+        }
+
+        return {
+            deleteBeforeReplace: deleteBeforeReplace,
+            replaces: replaces,
+            changes: changes
+        };
```

Comment:
can be simplified to just: `return { deleteBeforeReplace, replaces, changes };`

Reply:
Really? Those are keys in the return type. Could I still do what you are suggesting?

Reply:
yup.  in js/ts  `{ a, b, c }` is shorthand for `{ a: a, b: b, c: c }`.

Reply:
Ohh...learned something new!

Reply:
However, I'll leave it as-is in this case.

Reply:
Why? I agree it reads redundant.

Reply:
I'll leave it up to the reader to make any such changes since it is more of a code style optimization. I don't think the example needs to be as succinct as possible.